delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/model/StorageState.scala

```diff
@@ -81,17 +85,60 @@ object StorageState {
       updatedBy: Subject
   ) extends StorageState {
 
-    def storage: Storage =
+    def storage(cfg: StorageTypeConfig): Storage =
       value match {
-        case DiskStorageValue(default, algorithm, volume, read, write, maxSize)                       =>
-          DiskStorage(id, project, default, algorithm, volume, read, write, maxSize, tags, source)
-        case S3StorageValue(default, algorithm, bucket, settings, read, write, maxSize)               =>
-          S3Storage(id, project, default, algorithm, bucket, settings, read, write, maxSize, tags, source)
-        case RemoteDiskStorageValue(default, algorithm, endpoint, cred, folder, read, write, maxSize) =>
-          RemoteDiskStorage(id, project, default, algorithm, endpoint, cred, folder, read, write, maxSize, tags, source)
+        case DiskStorageValue(default, volume, read, write, maxSize)                                       =>
+          DiskStorage(
+            id = id,
+            project = project,
+            default = default,
+            algorithm = cfg.disk.digestAlgorithm,
+            volume = volume,
+            readPermission = read.getOrElse(cfg.disk.readPermission),
```

Comment:
It could be more interesting to pass the config to the STM in the next function so as to have it in the state instead of overwriting it here ?

Reply:
I tried this but in the there was a lot of extra code involved, since I had to pass the whole `storage: Storage`

Reply:
We can't have something like this in the STM ?
```
def next(storageConfig: StorageConfig)(state: StorageState, event: StorageEvent)
```
And maybe have a special state for the case of RemoteStorage/S3 storage which has been enabled and then disabled ?